vendor/bat-native-ads/src/bat/ads/history_item_info.cc

```diff
@@ -30,72 +28,37 @@ bool HistoryItemInfo::operator!=(const HistoryItemInfo& rhs) const {
   return !(*this == rhs);
 }
 
-std::string HistoryItemInfo::ToJson() const {
-  std::string json;
-  SaveToJson(*this, &json);
-  return json;
-}
-
-bool HistoryItemInfo::FromJson(const std::string& json) {
-  rapidjson::Document document;
-  document.Parse(json.c_str());
+base::Value::Dict HistoryItemInfo::ToValue() const {
+  base::Value::Dict dictionary;
 
-  if (document.HasParseError()) {
-    BLOG(1, helper::JSON::GetLastError(&document));
-    return false;
-  }
+  dictionary.Set("timestamp_in_seconds",
+                 base::NumberToString(created_at.ToDoubleT()));
+  dictionary.Set("ad_content", ad_content.ToValue());
+  dictionary.Set("category_content", category_content.ToValue());
+  return dictionary;
+}
 
-  if (document.HasMember("timestamp_in_seconds")) {
-    if (document["timestamp_in_seconds"].IsNumber()) {
-      // Migrate legacy timestamp
-      created_at =
-          base::Time::FromDoubleT(document["timestamp_in_seconds"].GetDouble());
-    } else {
-      double timestamp = 0.0;
-      if (base::StringToDouble(document["timestamp_in_seconds"].GetString(),
-                               &timestamp)) {
-        created_at = base::Time::FromDoubleT(timestamp);
-      }
+bool HistoryItemInfo::FromValue(const base::Value::Dict& root) {
+  if (const auto value = root.FindDouble("timestamp_in_seconds")) {
+    created_at = base::Time::FromDoubleT(*value);
```

Comment:
nit: Revert removal of `// Migrate legacy timestamp`

Reply:
Fixed in 0ba5b24424b1fcd4c09556aa630def61a121197f.